ci_framework/roles/polarion/tasks/main.yml

```diff
@@ -0,0 +1,110 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if variables are defined and not empty
+  ansible.builtin.assert:
+    that:
+      - cifmw_polarion_jump_repo_url is defined and cifmw_polarion_jump_repo_url != ""
+      - cifmw_polarion_test_id is defined and cifmw_polarion_test_id != ""
+      - cifmw_polarion_update_testcases is defined and cifmw_polarion_update_testcases | bool
+
+- name: Ensure deps are installed
+  become: true
+  tags:
+    - bootstrap
+    - packages
+  ansible.builtin.package:
+    name: "{{ cifmw_polarion_deps }}"
+    state: present
+
+- name: Look for test result XML files in artifacts directory
+  ansible.builtin.find:
+    paths: "{{ cifmw_polarion_jump_result_dir }}"
+    patterns: "*.xml"
+    recurse: yes
+  register: xml_files
+
+- name: Validate XML files with xmllint
+  ansible.builtin.command: "xmllint --noout {{ item.path }}"
+  loop: "{{ xml_files.files }}"
+  register: xmllint_output
+  changed_when: false
+  failed_when: xmllint_output.rc != 0
+
+- name: Deploy jump and report to polarion
+  when: xml_files.files | length > 0
+  block:
+  - name: Clone Jump repository
+    ansible.builtin.git:
+      repo: "{{ cifmw_polarion_jump_repo_url }}"
+      dest: "{{ cifmw_polarion_jump_repo_dir }}"
+      version: "{{ cifmw_polarion_jump_repo_branch }}"
+      force: true
+
+  - name: Create virtualenv
+    ansible.builtin.pip:
+      name: pip
+      virtualenv: "{{ cifmw_polarion_jump_repo_dir }}/jump-venv"
+      virtualenv_command: "/usr/bin/python3 -m venv"
+      state: latest
+      extra_args: --upgrade
+    when:
```

Comment:
care to move that `when:` right under the `name` of the task? It's usually easier to read.

Reply:
done